### DISCUSSION TITLE
How to use Server in V1.4.*

### DISCUSSION THREAD

**Discussion by daiguadaidai** [hyperium/hyper#2494] [Q&A] [answered]
hyper version: 1.4.*
in `lib.rs` file:
```
cfg_feature! {
    #![all(feature = "server")]

    pub mod server;
    #[cfg(any(feature = "http1", feature = "http2"))]
    #[doc(no_inline)]
    pub use crate::server::Server;
}
``` 
IDE`Clion+Rust Pulgin` can not import `Server`

hyper version: 1.3.*
```
pub use crate::server::Server;
...
pub mod server;
``` 
IDE can import `Server`

**Comment by davidbarsky** [comment, marked as answer]
Try declaring the `hyper` dependency in your `Cargo.toml` with the `full` feature flag:

```toml
hyper = { version = "0.14", features = ["full"] }
```

The documentation explains what `full` enables: https://docs.rs/hyper/0.14.5/hyper/#optional-features.

**Reply by sebpuetz** [reply]
I ran into the same issue while upgrading to `0.14`, if you don't want to pull in all features, it should be enough to specify `features = [ "server", "http1", "tcp"]` to make `Server::bind(&addr).serve(service)` compile. Looking at the feature gates in `lib.rs` any combination of `http1` or `http2` should suffice.
